wordle-web/src/scripts/wordsService.ts

```diff
@@ -7,9 +10,69 @@ export abstract class WordsService {
     return this.#words.includes(word)
   }
 
-  static validWords(): Array<string> {
-    //Todo
-    return new Array<string>()
+  static validWords(
+    game: WordleGame,
+    previousValid: string[] | undefined = undefined
+  ): Array<string> {
+    // all letters that were correct or misplaced
+    const validLetters = game.guesses.flatMap((guess) =>
+      guess.letters
+        .filter((l) => l.status == LetterStatus.Correct || l.status == LetterStatus.Misplaced)
+        .map((l) => l.char)
+    )
+    // all wrong letters
+    const wrongLetters = game.guesses.flatMap((guess) =>
```

Comment:
Not that this is a giant problem, but I might add in a simple check somewhere in this script to block the use of non -alpha characters from being submitted to the guess.